packages/melody-hooks/src/component.js

```diff
@@ -322,7 +327,14 @@ Object.assign(Component.prototype, {
             // Reset the hooks pointer
             this.hooksPointer = -1;
             // Run the component functions
-            this.data = this.componentFn(this.props) || {};
+
+            try {
+                const dataNext = this.componentFn(this.props) || {};
```

Comment:
Why do we need this temporary variable?

Reply:
Im not sure what would be assigned to `this.data` when the expression throws an error. When it still would have the same value after the error occurs, we can skip the variable. It was meant to avoid setting `this.data` to `undefined` in case of an error.

Reply:
Theoretically, if `this.componentFn` would throw upon execution, then the assignment to `this.data` would never be invoked, thus, `this.data` would still be assigned to its previous value:

```
let foo = 42;

function bar() {
  throw new Error('Hmm');
  return 7;
}

try {
  foo = bar();
} finally {
  console.log(foo); // prints "42"
}
```

Reply:
Interestingly, it seems like `try-finally` has been supported since ES3. It's just very rarely used.